### PR TITLE
[Merged by Bors] - chore: split `RingTheory/TensorProduct`

### PR DESCRIPTION
**Pull request by eric-wieser**
There is a reasonably-sized section on modules over two rings before we actual reach tensor products of rings.

The motivation for splitting here is to make room for the results in #6035.

The declarations are copied without modification, the module docstring has been adapted.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/TensorProduct/Tower.lean [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+ * `TensorProduct.AlgebraTensorModule.lift`
+ * `TensorProduct.AlgebraTensorModule.lift.equiv`
+ * `TensorProduct.AlgebraTensorModule.mk`
+ * `TensorProduct.AlgebraTensorModule.assoc`
+
+## Implementation notes
+
+We could thus consider replacing the less general definitions with these ones. If we do this, we
+probably should still implement the less general ones as abbreviations to the more general ones with
+fewer type arguments.
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/TensorProduct/Tower.lean:34]: This note is just copied from the other file: it's maybe still a good idea, but I didn't want to attempt it yet as it interferes with some other refactors I have in progress.

File: Mathlib/LinearAlgebra/TensorProduct/Tower.lean [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#align_import ring_theory.tensor_product from "leanprover-community/mathlib"@"88fcdc3da43943f5b01925deddaa5bf0c0e85e4e"
+
+/-!
+# The `A`-module structure on `M ⊗[R] N`
+
+When `M` is both an `R`-module and an `A`-module, and `Algebra R A`, then many of the morphisms
+preserve the actions by `A`.
+
+This file provides more general versions of the definitions already in
+`LinearAlgebra/TensorProduct`.
```
Comment (kbuzzard) [inline Mathlib/LinearAlgebra/TensorProduct/Tower.lean:18]: This is a directory, not a file. The directory contains one file which seems to be about matrices. Should this say something else?

**Comment by kbuzzard** [comment]
I have one trivial comment but this looks good to go.

File: Mathlib/LinearAlgebra/TensorProduct/Tower.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/LinearAlgebra/TensorProduct/Tower.lean:18]: Both `LinearAlgebra/TensorProduct/*` and `LinearAlgebra/TensorProduct.lean` exist. This is copied from the old docstring anyway, so I think fixing it is out of scope.

I think we'd like to be able to write `LinearAlgebra.TensorProduct` here (as in the import), but this is written with the hope that doc-gen will know to autolink paths to modules.

Comment (kbuzzard) [reply Mathlib/LinearAlgebra/TensorProduct/Tower.lean:18]: Aah, maybe `RingTheory/TensorProduct`?

Comment (kbuzzard) [reply Mathlib/LinearAlgebra/TensorProduct/Tower.lean:18]: Oh apologies, it is a directory *and* a file! I retract my comment.

**Comment by kbuzzard** [comment]
!maintainer merge

**Comment by eric-wieser** [comment]
maintainer merge

(! is only for bench)

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**